File: 2938-separate-black-and-white-balls/2938-separate-black-and-white-balls.c
#include <string.h>

/**
 * @brief Calculate the minimum number of steps to group all black balls to the right
 *        and all white balls to the left.
 *
 * @param s (char*): A binary string representing the sequence of black (1) and white (0) balls.
 *
 * @return (long long): The minimum number of swaps required.
 */
long long minimumSteps(char* s) {
    long long white_count = 0;  // Tracks the total number of swaps required
    long long min_steps = 0;    // Tracks the number of black balls encountered
    int n = strlen(s);          // Get the length of the string
    
    // Traverse the string character by character
    for (int i = 0; i < n; i++) {
        if (s[i] == '0') {
            // For each white ball, add the number of black balls encountered so far
            white_count += min_steps;
        } else {
            // For each black ball, increment the number of black balls encountered
            min_steps++;
        }
    }
    
    return white_count;  // Return the total number of swaps required
}

File: Daily Questions/2938-separate-black-and-white-balls/2938-separate-black-and-white-balls.py
class Solution:
    def minimumSteps(self, s: str) -> int:
        """
        Calculate the minimum number of steps to group all black balls to the right
        and all white balls to the left.

        Parameters:
        s (str): A binary string representing the sequence of black (1) and white (0) balls.

        Returns:
        int: The minimum number of swaps required.
        """
        white_count = 0  # Initialize the total number of swaps required
        min_steps = 0     # Track the number of black balls encountered so far
        
        for char in s:
            if char == '0':
                # For each white ball, add the number of black balls encountered so far
                # This simulates moving the black balls past the white balls
                white_count += min_steps
            else:
                # For each black ball, increment the number of black balls encountered
                min_steps += 1

        return white_count  # Return the total number of swaps required
